### DISCUSSION TITLE
How schedule works?

### DISCUSSION THREAD

**Discussion by ademar59** [renovatebot/renovate#29227] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

### Please tell us more about your question or problem

Hi,

we use schedule in packageRules:

```json
  "packageRules": [
    {
      "description": "create one MR for all minor,patch changes every month",
      "matchUpdateTypes": [
        "minor",
        "patch",
        "pin",
        "digest",
        "rollback"
      ],
      "groupName": "minor",
      "extends": ["schedule:monthly"]
    },
```

I will know how renovate do scheduling.
Need it to store the last execution ?
or
Just perform it when schedule period is reached ?
It means schedule:monthly = schedule:first day of the month

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment, marked as answer]
Renovate will create the PR when the schedule is satisfied. It is independent of the last run.

Make sure that the bot is run before 4am on the first day of the month. And other limits (say, prHourlyLimit, prConcurrentLimit) / conditions do not halt this PR creation, else you will have to wait till next month again :)

**Comment by rarkins** [comment]
https://docs.renovatebot.com/key-concepts/scheduling/

**Comment by ademar59** [comment]
Thanks for your explains

Because we have a lot of repos we have chosen to run the bot only for 0 8,12,16,20 * * 1-5
It means weekday and not before 4am
It mean that doesnt works if the first day of the month is saturday or sunday :(

I will change it to have a bot scheduling according to the preset schedules.
